ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -145,8 +145,13 @@ export function useTransactionDisplayData (transactionGroup) {
       : initialTransaction.sourceTokenSymbol
     primaryDisplayValue = swapTokenValue
     secondaryDisplayValue = swapTokenFiatAmount
-    prefix = isViewingReceivedTokenFromSwap ? '+' : '-'
-
+    if (isNegative) {
+      prefix = ''
+    } else if (isViewingReceivedTokenFromSwap && !isNegative) {
+      prefix = '+'
+    } else {
+      prefix = '-'
+    }
```

Comment:
I wrote out the four different branches to help myself think through it:
```suggestion
    if (isViewingReceivedTokenFromSwap) {
      if (isNegative) {
        // do nothing - the minus should already be there
      } else {
        prefix = '+'
      }
    } else {
      if (isNegative) {
        // this case should be impossible - how can you send _negative_ tokens?
      } else {
        // a positive value should be shown as a loss, as this is the amount of tokens sent.
        prefix = '-'
      }
    }
```

The only one you're missing is where a negative amount of tokens was sent, and I can't imagine how that would be possible. So I think your logic here is sound.